pkgs/os-specific/linux/nvidia-x11/generic.nix

```diff
@@ -30,7 +30,7 @@ assert ! versionOlder version "391" -> stdenv.hostPlatform.system == "x86_64-lin
 let
   nameSuffix = optionalString (!libsOnly) "-${kernel.version}";
   pkgSuffix = optionalString (versionOlder version "304") "-pkg0";
-  i686bundled = versionAtLeast version "391";
+  i686bundled = stdenv.hostPlatform.system == "i686-linux" && versionAtLeast version "391";
```

Comment:
If i remember correctly, 391 is the first driver, where nvidia has dropped support for 32bit driver, and the 32bit libraries for steam games and i686 programs run on x64 are bundled in the x64 package.